libipl/p10/ipl0.C

```diff
@@ -387,9 +388,27 @@ static void process_guard_records()
 		return;
 	}
 
-	openpower::guard::libguard_init(false);
+	openpower::guard::GuardRecords records;
+	try {
+		openpower::guard::libguard_init(false);
+		records = openpower::guard::getAll();
+	} catch (const openpower::guard::exception::InvalidGuardFile &ex) {
```

Comment:
Any exception from Guard lib, skips the guard function , This should be informed to user via common mechanism.  In  summary need to log error in both cases. luke devender sugested earlier.

Reply:
Jayanth, we discussed this extensively yesterday and we will be fixing the specific issue with this fix since we are in MF. There are two issues opened for handling other scenarios. For now we would just catch the Invalidguardfile exception and create a PEL. For others, its just a trace because we need to figure out what all exceptions to catch which is not the scope of this defect.

Reply:
What is the side effect on logging error and continuing boot for all guard exceptions here .  Here both case skipping guard functionalities and continuing ipl , One case logging error and other case just logging message.  i would prefer to be consistent now. Incase you are planing to actual solution later add todo with github issue mentioned .

Reply:
The PEL that we are using is a InvalidGuardFile PEL with the description "empty or no guard file". And it may not be appropriate one to use on all  exceptions. Creating another PEL seems out of the scope of this defect. So we wanted to address it later.